ple/games/donkeykong/Fireball.py

```diff
@@ -0,0 +1,116 @@
+        self.__speed = speed
+
+    #Update the image of a fireball
+    def updateImage(self, raw_image):
+        self.image = raw_image
+        self.image = pygame.transform.scale(self.image, (15, 15))
+
+    #Getters and Setters for some private variables
+    def getSpeed(self):
+        return self.__speed
+    def setSpeed(self,speed):
+        self.__speed = speed
+    def getFall(self):
+        return self.__fall
+    def getDirection(self):
+        return self.__direction
+
+    #Moves the fireball in the required direction
+    def continuousUpdate(self, wallGroup, ladderGroup):
+
+        #The fireball is falling
+        if self.__fall == 1:
+            #We move the fireball downwards with speed of self.__speed
+            self.update(self.image, "V", self.__speed)
+            if self.checkCollision(wallGroup, "V"):
+                #We have collided with a wall below, so the fireball can stop falling
+                self.__fall = 0
+                #Set the direction randomly
+                self.__direction = int(math.floor(random.random() * 100)) % 2
+
+        else:
+
+            #While we are on the ladder, we use a probability of 4/20 to make the fireball start falling
+            if self.checkCollision(ladderGroup, "V") and len(self.checkCollision(wallGroup, "V")) == 0:
+                randVal = int(math.floor(random.random() * 100)) % 20
+                if randVal < 15:
+                    self.__fall = 0
+                else:
+                    self.__fall = 1
+
+            #We are at the edge of the floor so the fireball starts falling
+            if len(self.checkCollision(ladderGroup, "V")) == 0 and len(self.checkCollision(wallGroup, "V")) == 0:
+                self.__fall = 1
+
+            #We are moving right, so update the fireball image to the right
+            if self.__direction == 0:
+                self.update(pygame.image.load('Assets/fireballright.png'), "H", self.__speed)
+                #When we hit a wall, we change direction
+                if self.checkCollision(wallGroup, "H"):
+                    self.__direction = 1
+                    self.update(self.image, "H", -self.__speed)
+
+            #We are moving left, so update the fireball image to the left
+            else:
+                self.update(pygame.image.load('Assets/fireballleft.png'), "H", -self.__speed)
+                #When we hit a wall, we change direction
+                if self.checkCollision(wallGroup, "H"):
+                    self.__direction = 0
+                    self.update(self.image, "H", self.__speed)
+
+    #Move the fireball in the required direction with the required value and also set the image of the fireball
+    def update(self, raw_image, direction, value):
+        if direction == "H":
+            self.setPosition((self.getPosition()[0] + value, self.getPosition()[1]))
+            self.image = raw_image
+            self.image = pygame.transform.scale(self.image, (15, 15))
```

Comment:
is it possible to cache the scaled down image? instead of performing it at each update call?